.github/workflows/tests.yml

```diff
@@ -0,0 +1,89 @@
+name: CI tests
+
+defaults:
+  run:
+    shell: bash
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+
+    container: 
+        image: koreader/koappimage:0.2.2
+        options: --user ko
+        
+    env:
+        EMULATE_READER: 1
+        # this is for shellcheck 0.4.5 and lower; can be removed for 0.4.6
+        LC_ALL: en_US.UTF8
+        HOME: /home/ko
+        # pretend to be cirleci so the scripts dont have to change
+        CIRCLE_WORKING_DIRECTORY: ${{ github.workspace }}
+
+    steps:
+        # chown working directory to ko (docker user0
+      - name: permissions
+        run: sudo chown -R ko:ko /__w/koreader/
+
+        # make sure we have git >= 2.18 so github checkout behaves
```

Comment:
lol, here's what I wrote when I played around with it a bit last month but I couldn't quite get the permissions working properly:

```yaml
      # Because we need at least 2.18 for reasons…
      # Cf. <https://github.com/actions/checkout/issues/335>
      - name: Force install latest Git
        run: |
          sudo apt-get update \
          && sudo apt-get install -y software-properties-common \
          && sudo add-apt-repository -y ppa:git-core/ppa \
          && sudo apt-get update \
          && sudo apt-get install -y git
      - name: Cleaning up the $GITHUB_WORKSPACE as root from a Docker image
        # Volume auto mounted by gh actions pointing to the current working-directory
        run: find /__w/${{ github.event.repository.name }}/${{ github.event.repository.name }}/. -name . -o -prune -exec rm -rf -- {} + || true

      # Nonsense due to  https://github.com/actions/runner/issues/691
      - name: Change dir owner to working user
        run: sudo chown -R $USER:$USER $GITHUB_WORKSPACE
```